### ISSUE TITLE
TextType empty_data = null if string is empty

### ISSUE THREAD

**Issue by markoffk** [symfony/symfony-docs#7596]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Symfony version  | 3.2.4

Hello. 

I am getting null value after passing empty string to symfony form field. 
After defining empty_data in form builder, the problem has gone:
```
        $builder
            ->add('name', TextType::class, [
                'empty_data'  => '',
            ])
```

But documentation claims that empty_data is empty string ('') by default. So either the documentation should be changed or behavior of form field type.

Link to docs: http://symfony.com/doc/current/reference/forms/types/text.html#empty-data

Thanks.

**Comment by HeahDude** [comment]
The feature has been introduced in 3.1 in https://github.com/symfony/symfony/pull/18357, but the target should now be 3.2. It would be easier to handle it once https://github.com/symfony/symfony-docs/pull/6265 is merged.

**Comment by curry684** [comment]
Why was this put on hold? More people are running into this, see #22192

**Comment by HeahDude** [comment]
Hi @curry684, I've changed the label, see #https://github.com/symfony/symfony-docs/pull/6265, `empty_data` option needs a revamp, and some more work on symfony 2.7. I've changed the label to better indicate the status.

**Comment by canaaerus** [comment]
I have just encounterd this bug. Is it a problem in the documentation or do you intent to change the behavior of the TextType?

**Comment by HeahDude** [comment]
The behavior is now `null` by default or empty string if explicitly configured (which not possible before https://github.com/symfony/symfony/pull/18357), this cannot change for BC.

But it may be the opposite in 4.0, I'm working on it (no PR is open yet) but this is not as simple as it sounds :).

**Comment by xabbuh** [comment]
We need to update the `empty_data` option description for the text type to reflect that the default is `null`, but that you can opt-in for the empty string (see symfony/symfony#18357).

**Comment by xabbuh** [comment]
see also symfony/symfony#29061 for problems one runs into when reading the current description

**Comment by HeahDude** [comment]
Fixed in #10743. Closing here, thanks for the report.
